contrib/errorprone/src/python/pants/contrib/errorprone/tasks/errorprone.py

```diff
@@ -0,0 +1,129 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+from pants.backend.jvm.subsystems.shader import Shader
+from pants.backend.jvm.targets.jar_dependency import JarDependency
+from pants.backend.jvm.tasks.nailgun_task import NailgunTask
+from pants.base.exceptions import TaskError
+from pants.base.workunit import WorkUnitLabel
+from pants.util.dirutil import safe_mkdir
+from pants.util.strutil import safe_shlex_split
+
+
+class ErrorProne(NailgunTask):
+  """Check Java code for Error Prone violations.  See http://errorprone.info/ for more details."""
+
+  _ERRORPRONE_MAIN = 'com.google.errorprone.ErrorProneCompiler'
+  _JAVA_SOURCE_EXTENSION = '.java'
+
+  @classmethod
+  def register_options(cls, register):
+    super(ErrorProne, cls).register_options(register)
+
+    register('--skip', type=bool, help='Skip Error Prone.')
+    register('--transitive', default=False, type=bool,
+             help='Run Error Prone against transitive dependencies of targets '
+                  'specified on the command line.')
+    register('--excluded-targets', type=list, default=[], fingerprint=True,
+             help='Targets to be excluded from analysis.')
+    register('--command-line-options', type=list, default=[], fingerprint=True,
+             help='Command line options passed to Error Prone')
+
+    cls.register_jvm_tool(register,
+                          'errorprone',
+                          classpath=[
+                            JarDependency(org='com.google.errorprone',
+                                          name='error_prone_core',
+                                          rev='2.0.15'),
+                          ],
+                          main=cls._ERRORPRONE_MAIN,
+                          custom_rules=[
+                            Shader.exclude_package('com.google.errorprone', recursive=True),
+                          ]
+                         )
+
+  @classmethod
+  def prepare(cls, options, round_manager):
+    super(ErrorProne, cls).prepare(options, round_manager)
+    round_manager.require_data('runtime_classpath')
+
+  def _is_errorprone_target(self, target):
+    return target.has_sources(self._JAVA_SOURCE_EXTENSION) and not target.is_synthetic
```

Comment:
This method was basically lifted from checkstyle.py which seems to be working.  What resources are you thinking need to be excluded?